pkg/validation/variables/variables_component.go

```diff
@@ -189,6 +195,81 @@ func validateAndReplaceForOpenShiftComponent(variables map[string]string, opensh
 	return newInvalidKeysError(invalidKeys)
 }
 
+// validateAndReplaceForImageComponent validates the image component data for global variable references and replaces them with the variable value
+func validateAndReplaceForImageComponent(variables map[string]string, image *v1alpha2.ImageComponent) error {
+	var err error
+
+	invalidKeys := make(map[string]bool)
+
+	if image != nil {
+		// Validate image's image name
+		if image.ImageName, err = validateAndReplaceDataWithVariable(image.ImageName, variables); err != nil {
+			checkForInvalidError(invalidKeys, err)
+		}
+
+		switch {
+		case image.Dockerfile != nil:
+			if err = validateAndReplaceForDockerfileImageComponent(variables, image.Dockerfile); err != nil {
+				checkForInvalidError(invalidKeys, err)
+			}
+		}
+
+	}
+
+	return newInvalidKeysError(invalidKeys)
+}
+
+// validateAndReplaceForDockerfileImageComponent validates the dockerfile image component data for global variable references and replaces them with the variable value
+func validateAndReplaceForDockerfileImageComponent(variables map[string]string, dockerfileImage *v1alpha2.DockerfileImage) error {
+	var err error
```

Comment:
please consider getting rid of redundant nesting, like
```go
if dockerfileImage == nil {
  return null
}

var err error
invalidKeys := make(map[string]bool)
switch {
case dockerfileImage.Uri != "":
...
return newInvalidKeysError(invalidKeys)
```

Reply:
updated for the entire pkg and updated tests for coverage